From 57cc17c050512b78778fcf8e49655b42d6c57e80 Mon Sep 17 00:00:00 2001
From: oddcoder <ahmedsoliman@oddcoder.com>
Date: Sat, 22 Feb 2020 18:26:46 +0200
Subject: [PATCH] Stage 1 with debug info

---
 Makefile | 52 +++++++++++++++++++++-------------------------------
 1 file changed, 21 insertions(+), 31 deletions(-)

diff --git a/Makefile b/Makefile
index 97c4ebe..2f1b4d5 100644
--- a/Makefile
+++ b/Makefile
@@ -5,28 +5,9 @@ CARGOFLAGS=
 RUSTCFLAGS=
 
 BUILD=target
-ifneq ($(TARGET),)
-	BUILD="target/$(TARGET)"
-	CARGOFLAGS="--target=$(TARGET)"
-endif
-
-ifeq ($(TARGET),aarch64-unknown-linux-gnu)
-	export CC=aarch64-linux-gnu-gcc
-	export LD=aarch64-linux-gnu-ld
-	export AR=aarch64-linux-gnu-ar
-endif
-
-ifeq ($(TARGET),aarch64-unknown-redox)
-	export CC=aarch64-unknown-redox-gcc
-	export LD=aarch64-unknown-redox-ld
-	export AR=aarch64-unknown-redox-ar
-endif
-
-ifeq ($(TARGET),x86_64-unknown-redox)
-	export CC=x86_64-unknown-redox-gcc
-	export LD=x86_64-unknown-redox-ld
-	export AR=x86_64-unknown-redox-ar
-endif
+export CC=x86_64-lfs-linux-relibc-gcc
+export LD=x86_64-lfs-linux-relibc-ld
+export AR=x86_64-lfs-linux-relibc-ar
 
 SRC=\
 	Cargo.* \
@@ -66,6 +47,15 @@ libs: \
 	$(BUILD)/release/crtn.o \
 	$(BUILD)/release/ld_so
 
+libs-debug: \
+	$(BUILD)/debug/libc.a \
+	$(BUILD)/debug/libc.so \
+	$(BUILD)/debug/crt0.o \
+	$(BUILD)/debug/crti.o \
+	$(BUILD)/debug/crtn.o \
+	$(BUILD)/debug/ld_so
+
+
 install-libs: libs
 	mkdir -pv "$(DESTDIR)/lib"
 	cp -v "$(BUILD)/release/libc.a" "$(DESTDIR)/lib"
@@ -79,15 +69,15 @@ install-libs: libs
 
 install: install-headers install-libs
 
-gentoo-install-libs: libs
+gentoo-install-libs: libs-debug
 	mkdir -p "$(TMP)/$(LIB)"
 	mkdir -p "$(TMP)/usr/$(LIB)"
-	cp -v "$(BUILD)/release/libc.a" "$(TMP)/usr/$(LIB)"
-	cp -v "$(BUILD)/release/libc.so" "$(TMP)/$(LIB)"
-	cp -v "$(BUILD)/release/crt0.o" "$(TMP)/usr/$(LIB)"
-	cp -v "$(BUILD)/release/crti.o" "$(TMP)/usr/$(LIB)"
-	cp -v "$(BUILD)/release/crtn.o" "$(TMP)/usr/$(LIB)"
-	cp -v "$(BUILD)/release/ld_so" "$(TMP)/$(LIB)/ld64.so.1"
+	cp -v "$(BUILD)/debug/libc.a" "$(TMP)/usr/$(LIB)"
+	cp -v "$(BUILD)/debug/libc.so" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/debug/crt0.o" "$(TMP)/usr/$(LIB)"
+	cp -v "$(BUILD)/debug/crti.o" "$(TMP)/usr/$(LIB)"
+	cp -v "$(BUILD)/debug/crtn.o" "$(TMP)/usr/$(LIB)"
+	cp -v "$(BUILD)/debug/ld_so" "$(TMP)/$(LIB)/ld64.so.1"
 	cp -v "$(BUILD)/openlibm/libopenlibm.a" "$(TMP)/usr/$(LIB)/libm.a"
 	cp -v "$(BUILD)/pthreads-emb/libpthread.a" "$(TMP)/usr/$(LIB)/libpthread.a"
 
@@ -99,9 +89,9 @@ gentoo-install-headers: headers
 	cp -v "openlibm/src"/*.h "$(TMP)/usr/include"
 	cp -v "pthreads-emb/"*.h "$(TMP)/usr/include"
 
-
 gentoo-install: gentoo-install-headers gentoo-install-libs
 
+
 sysroot: all
 	rm -rf $@
 	rm -rf $@.partial
@@ -162,7 +152,7 @@ $(BUILD)/release/libc.a: $(BUILD)/release/librelibc.a $(BUILD)/pthreads-emb/libp
 	$(AR) -M < "$@.mri"
 
 $(BUILD)/release/libc.so: $(BUILD)/release/librelibc.a $(BUILD)/pthreads-emb/libpthread.a $(BUILD)/openlibm/libopenlibm.a
-	$(CC) -nostdlib -shared -Wl,--allow-multiple-definition -Wl,--whole-archive $^ -Wl,--no-whole-archive -Wl,-soname,libc.so.6 -o $@
+	$(CC) -nostdlib -shared -Wl,--allow-multiple-definition -Wl,--whole-archive $^ -Wl,--no-whole-archive -o $@
 
 $(BUILD)/release/librelibc.a: $(SRC)
 	CARGO_INCREMENTAL=0 $(CARGO) rustc --release $(CARGOFLAGS) -- --emit link=$@ $(RUSTCFLAGS)
-- 
2.24.1

